lib/tokens.js

```diff
@@ -0,0 +1,687 @@
+ * to ensure they have not changed but will be readable by any external party
+ * that sees the token. For this reason, these attributes should never include
+ * uniquely identifying information about the entity to which the tokens are
+ * linked (as this would defeat the tokenization scheme).
+ *
+ * Tokens may be "pairwise resolved" *just once* to *one external party*. See
+ * the `resolve` function for more details.
+ *
+ * @param {object} options - Options to use.
+ * @param {object} options.tokenizer - The tokenizer instance to use to
+ *   create the tokens.
+ * @param {string} options.internalId - The internal ID the tokens will be
+ *   linked to.
+ * @param {Uint8Array} [options.attributes] - Attributes that will be encoded
+ *   in each token such that they can be authenticated but also such that
+ *   they will appear in the clear for users of the token. Attributes are
+ *   expected to be encoded in an application-specific way to fit within a
+ *   maximum of 8 bytes.
+ * @param {number} options.tokenCount - The number of tokens to create
+ *   in this call; note that this is NOT the batch size and that created tokens
+ *   may belong to a number of different token batches.
+ *
+ * @returns {object} An object with `tokens` as an array of created tokens.
+ */
+export async function create(
+  {internalId, attributes = new Uint8Array(), tokenCount} = {}) {
+  // TODO: validate `tokenCount`
+  if(!(attributes instanceof Uint8Array)) {
+    throw new TypeError('"attributes" must be a Uint8Array.');
+  }
+  if(attributes.length > MAX_AAD_SIZE) {
+    throw new TypeError(`"attributes" maximum size is ${MAX_AAD_SIZE} bytes.`);
+  }
+
+  // get the current tokenizer and its HMAC API
+  const {id: tokenizerId, hmac} = await tokenizers.getCurrent();
+
+  // get version associated with tokenizer, creating it as needed
+  const {tokenVersion} = await tokenVersions.ensureTokenVersion(
+    {tokenizerId});
+
+  // create tokens with limited concurrency
+  // TODO: determine if this is a desirable limitation or if another
+  // approach is better
+  const {tokenCreationConcurrency: concurrency} = bedrock.config.tokenization;
+  const limit = pLimit(concurrency);
+
+  // create tokens until `tokenCount` is reached
+  const tokens = [];
+  while(tokens.length < tokenCount) {
+    // 1. Calculate remaining tokens to issue.
+    const target = tokenCount - tokens.length;
+
+    // 2. Get an open batch for creating tokens.
+    const {tokenBatch, startIndex, claimedTokenCount} = await _getOpenBatch(
+      {internalId, tokenVersion, tokenCount: target});
+
+    // 3. Create tokens in parallel with concurrency limit.
+    const promises = [];
+    for(let i = 0; i < claimedTokenCount; ++i) {
+      const index = startIndex + i;
+      promises.push(limit(() => _createToken(
+        {hmac, tokenVersion, tokenBatch, index, attributes})));
+    }
+    const newTokens = await Promise.all(promises);
```

Comment:
Some of the other promise-fun tools could help here.  Perhaps https://github.com/sindresorhus/p-times is shortest and drops pLimit and the loop (untested):
```suggestion
    const newTokens = await pTimes(claimedTokenCount,
      i => _createToken({hmac, tokenVersion, tokenBatch, startIndex + i, attributes}),
      {concurrency});
```